reference-implementation/lib/helpers.js

```diff
@@ -62,6 +62,10 @@ exports.IsFiniteNonNegativeNumber = v => {
 };
 
 exports.InvokeOrNoop = (O, P, args) => {
+  assert(exports.typeIsObject(O) === true);
+  assert(typeof P === 'string');
+  assert(Array.isArray(args));
```

Comment:
The helpers were already a bit inconsistent so it's just InvokeOrNoop we'd remove it from; `PromiseInvokeOrFallbackOrNoop` never created empty Lists if `args1` or `args2` were undefined.